src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1551,34 +1551,32 @@ private static BoundBlock BindMethodBody(MethodSymbol method, TypeCompilationSta
                     body = (BoundBlock)binder.BindEmbeddedBlock(blockSyntax, diagnostics);
                     importChain = binder.ImportChain;
 
-                    if (method.MethodKind == MethodKind.Destructor)
+                    if (method.MethodKind != MethodKind.Destructor)
                     {
-                        return MethodBodySynthesizer.ConstructDestructorBody(method, body);
-                    }
-
-                    foreach (var iterator in binder.MethodSymbolsWithYield)
-                    {
-                        foreach (var parameter in iterator.Parameters)
+                        foreach (var iterator in binder.MethodSymbolsWithYield)
                         {
```

Comment:
I think this loop should be run for all methods, destructor or not (I realize it wasn't before).